### DISCUSSION TITLE
Any chance of making this work on android?

### DISCUSSION THREAD

**Discussion by NishantJh** [nefarius/DsHidMini#179] [Q&A]
Hi, I use DsHidMini on my PC to make DS3 controllers work and they work beautifully for every game that uses controllers. Thanks to the developer for that. But now, for Forza Horizon 5, I tried to GameStream the game to my android TV Box to play the game on my TV. But the controller does not work as it is recognized as a DS3, which is not supported by Forza Horizon. Any way to make it work on Android as it works on Windows?

**Comment by nefarius** [comment]
Hello! I don't understand your setup. So you connect your DS3 to your Android device? That streams the game from PC?

**Reply by NishantJh** [reply]
Yes. NVIDIA has a Gamestream that allows this setup. So the game runs on the PC, but controller is connected to android device.

**Reply by nefarius** [reply]
And how is the controller "forwarded" to your PC? Does NVIDIA emulate like an X360 or DS4 pad or similar?

**Reply by NishantJh** [reply]
Not sure what NVIDIA does in the background. Gamestream and Steam's Remote Play both allow controllers to be connected to remote devices. For Gamestream, there is an app called Moonlight on android that connects to Gamestream and for Steam there an app called Steam link. Those apps seem to be sending the Controller inputs to PC.

**Reply by NishantJh** [reply]
Even Xbox has a similar feature called Remote Play.

**Reply by NishantJh** [reply]
Apparently there is a software https://www.virtualhere.com/ that converts USB connected devices on remote systems work like they were connected to local PC. Haven't tried if this would work, but could help you think of something.

**Reply by nefarius** [reply]
I know how Moonlight works so I guess it's the same for your Android setup. It emulates a standard well-known gamepad on your PC and gets the inputs over network. For that to work with your DS3 the Android device would need to run a driver or service or similar that can interface with the DS3 and apply the same compatibility efforts like DsHidMini does. I'm afraid I am not aware of any Android solution and I am certainly the wrong guy to ask about Android stuff 😅 So yeah, probably better to use a modern controller in this scenario.

**Reply by nefarius** [reply]
> Apparently there is a software https://www.virtualhere.com/ that converts USB connected devices on remote systems work like they were connected to local PC. Haven't tried if this would work, but could help you think of something.

If it presents the DS3 as the same USB device as it would when directly connected, DsHidMini will most probably simply work on that too. Experiment, I have no experience with your setup.

**Reply by NishantJh** [reply]
> 

Even my DS4 has the same issue as Forza needs Xbox Controller. Thanks for your reply. Guess I'll have to buy an Xbox controller. Forza Horizon looks amazing on the TV screen, enough to distract from the race itself  :p

**Reply by nefarius** [reply]
Forza Horizon 5 is an amazing game, currently dumping hours into it. I guess you're restricted to what the whole NVIDIA solution supports and most probably the easiest way is to go full compatibility with an Xbox Controller. Or alternatively, would it also work if you just connect the DS3 to your PC via Bluetooth or is it too far away?
